tests/integration/test_charm.py

```diff
@@ -6,27 +6,204 @@
+                    helpers.ZOOKEPER_NAME,
+                )
+                await ops_test.model.wait_for_idle(
+                    apps=[helpers.POSTGRES_NAME, helpers.CERTIFICATES_NAME, helpers.ZOOKEPER_NAME],
+                    status="active",
+                    timeout=10 * 60,
+                )
+
+                logger.info(
+                    "Waiting for '%s, %s' to settle into 'blocked-idle'", helpers.KAFKA_NAME, helpers.OPENSEARCH_NAME
+                )
+                await ops_test.model.wait_for_idle(
+                    apps=[helpers.KAFKA_NAME, helpers.OPENSEARCH_NAME],
+                    status="blocked",
+                    raise_on_blocked=False,
+                    timeout=30 * 60,
+                )
+
+            # Relate the dependencies
+            logger.info("Integrating dependencies")
+            await asyncio.gather(
+                ops_test.model.integrate(helpers.KAFKA_NAME, helpers.ZOOKEPER_NAME),
+                ops_test.model.integrate(helpers.OPENSEARCH_NAME, helpers.CERTIFICATES_NAME),
+            )
+
+            # Wait for the dependencies to settle
+            logger.info("Waiting for dependencies to settle into 'active-idle'")
+            await ops_test.model.wait_for_idle(
+                apps=[
+                    helpers.KAFKA_NAME,
+                    helpers.OPENSEARCH_NAME,
+                    helpers.POSTGRES_NAME,
+                    helpers.CERTIFICATES_NAME,
+                    helpers.ZOOKEPER_NAME,
+                ],
+                status="active",
+                timeout=15 * 60,
+            )
+
+            # Create offers for DataHub to consume
+            logger.info("Creating offers")
+            await asyncio.gather(
+                ops_test.model.create_offer(
+                    endpoint="kafka-client", application_name=helpers.KAFKA_NAME, offer_name=helpers.KAFKA_OFFER_NAME
+                ),
+                ops_test.model.create_offer(
+                    endpoint="opensearch-client",
+                    application_name=helpers.OPENSEARCH_NAME,
+                    offer_name=helpers.OPENSEARCH_OFFER_NAME,
+                ),
+                ops_test.model.create_offer(
+                    endpoint="database", application_name=helpers.POSTGRES_NAME, offer_name=helpers.POSTGRES_OFFER_NAME
+                ),
+            )
+
+        with ops_test.model_context(k8s_model):
+            async with ops_test.fast_forward(fast_interval="5m"):
+                # Deploy DataHub
+                logger.info("Deploying '%s'", helpers.APP_NAME)
+                await helpers.deploy_charm(ops_test, charm)
+
+                # TODO (mertalpt): Find a way to avoid doing this.
+                # Gives `pebble_ready` time to run before making relations.
+                # I suspect it is a problem because gigabytes of OCI images are not cached
+                # and take a long time to be downloaded.
+                await asyncio.sleep(10 * 60)
```

Comment:
could you add this as `idle_period` in the `wait_for_idle` below?
https://pythonlibjuju.readthedocs.io/en/latest/api/juju.model.html#juju.model.Model.wait_for_idle

Reply:
It is not really the same thing. That option checks for continuous `idle` time. This is to allow images to get downloaded and `pebble_ready` to run. I intend to re-do the hooks (esp. for `update-status`) and the tests. This should be better afterwards.